integration_test/upgrade_test.go

```diff
@@ -14,4 +14,44 @@
 
 package integrationtest
 
-// TODO(ahmetb): implement upgrade tests (https://krew.dev/issues/233)
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"sigs.k8s.io/krew/pkg/constants"
+)
+
+func TestKrewUpgrade(t *testing.T) {
+	skipShort(t)
+
+	test, cleanup := NewTest(t)
+	defer cleanup()
+
+	test.WithIndex().
+		Krew("install", "--manifest", filepath.Join("testdata", validPlugin+constants.ManifestExtension)).
+		RunOrFail()
+	initialLocation := realLocation(test, validPlugin)
+
+	test.Krew("upgrade").RunOrFail()
+	eventualLocation := realLocation(test, validPlugin)
+
+	if initialLocation == eventualLocation {
+		t.Errorf("Expecting the plugin path to change but was the same.")
+	}
+}
+
+func realLocation(test *ITest, plugin string) string {
```

Comment:
resolvePluginSymlink is a better name.
"real" has different connotations like realpath(2).

Reply:
Yes, so true..